src/decode/lzma.rs

```diff
@@ -151,136 +144,86 @@ where
     rep_len_decoder: rangecoder::LenDecoder,
 }
 
-// Initialize decoder with accumulating buffer
-pub fn new_accum<W>(
-    output: lzbuffer::LzAccumBuffer<W>,
-    lc: u32,
-    lp: u32,
-    pb: u32,
-    unpacked_size: Option<u64>,
-) -> DecoderState<W, lzbuffer::LzAccumBuffer<W>>
-where
-    W: io::Write,
-{
-    DecoderState {
-        _phantom: PhantomData,
-        partial_input_buf: std::io::Cursor::new([0; MAX_REQUIRED_INPUT]),
-        output,
-        lc,
-        lp,
-        pb,
-        unpacked_size,
-        literal_probs: vec![vec![0x400; 0x300]; 1 << (lc + lp)],
-        pos_slot_decoder: vec![rangecoder::BitTree::new(6); 4],
-        align_decoder: rangecoder::BitTree::new(4),
-        pos_decoders: [0x400; 115],
-        is_match: [0x400; 192],
-        is_rep: [0x400; 12],
-        is_rep_g0: [0x400; 12],
-        is_rep_g1: [0x400; 12],
-        is_rep_g2: [0x400; 12],
-        is_rep_0long: [0x400; 192],
-        state: 0,
-        rep: [0; 4],
-        len_decoder: rangecoder::LenDecoder::new(),
-        rep_len_decoder: rangecoder::LenDecoder::new(),
+impl DecoderState {
+    // Initialize decoder with accumulating buffer
+    pub fn new(lc: u32, lp: u32, pb: u32, unpacked_size: Option<u64>) -> Self {
```

Comment:
Rather than passing the parameters one by one to the function, could this API use `LzmaParams` struct? This would avoid potential mistakes in swapping `lc`/`lp`/`pb` between the call site and the function definition. Or at the very least define a new `LzmaProperties` struct for `(lc, lp, pb)`, and use that for all functions that take as input these 3 parameters together.

Reply:
For now I've defined an `LzmaProperties` struct for `lclppb`. There may be a cleaner way of exposing this but this can be put off for bikeshedding a public raw decoder API.